components/director/internal/domain/formation/resolver.go

```diff
@@ -352,3 +352,86 @@ func (r *Resolver) FormationAssignment(ctx context.Context, obj *graphql.Formati
 
 	return r.formationAssignmentConv.ToGraphQL(formationAssignment)
 }
+
+// Status retrieves a Status for the specified Formation
+func (r *Resolver) Status(ctx context.Context, obj *graphql.Formation) (*graphql.FormationStatus, error) {
+	param := dataloader.ParamFormationStatus{ID: obj.ID, Ctx: ctx}
+	return dataloader.FormationStatusFor(ctx).FormationStatusByID.Load(param)
+}
+
+// StatusDataLoader retrieves a Status for each Formation ID in the keys argument
+func (r *Resolver) StatusDataLoader(keys []dataloader.ParamFormationStatus) ([]*graphql.FormationStatus, []error) {
+	if len(keys) == 0 {
+		return nil, []error{apperrors.NewInternalError("No Formations found")}
+	}
+
+	ctx := keys[0].Ctx
+	formationIDs := make([]string, 0, len(keys))
+	for _, key := range keys {
+		formationIDs = append(formationIDs, key.ID)
+	}
+
+	var ( // These are the default values for listing
+		cursor = ""
+		first  = 200
```

Comment:
Setting a maximum amount of FAs that can be retrieved per formation is incorrect and may lead to invalid behaviour.

With the current implementation, it is assumed that each formation must have maximum 200 FAs which is not guaranteed at all. Assume that given formation has 201 FAs for the sake of the example -> 200 are in status READY and 1 is in status ERROR -> we will get the 200 READY and will assume formation status READY which is invalid.

We should change how we retrieve the FAs. I assume you kept the `cursor` and `first` because they are required for the `unionLister` that supports paging, but in this case we won't need any paging.

So, in the FA repository a new func has to added that retrieves all FAs for the given formationIDs with `IN` clause instead of a `UNION` query:

1. `select all FAs where formation_id IN (<formation IDs slice>)`
2. then map them accordingly (FAs per Formation) and return them
3. then in the resolver the status is calculated based on all FAs